### PR TITLE
feat(helpers)!: use const generics where possible

### LINKED ISSUE #2679: Support `const` modifier on type parameters
### Clear and concise description of the problem

As a developer using faker I want to benefit from `const` modifier on type parameters so that I don't need to explicitly add `as const`.

### Suggested solution

Add `const` modifier on type parameters.

### Alternative

_No response_

### Additional context

This feature was introduced by TypeScript v5.0.0 and would break TypeScript support < v5

### PR DESCRIPTION
**Pull request by ST-DDT**
Implements #2679

- #2679

This feature requires TS5+ and thus is a breaking change.

### CODE REVIEW

**Comment by Shinigami92** [comment]
Should we add a https://vitest.dev/api/expect-typeof.html for this specific case?

**Comment by ST-DDT** [comment]
> Should we add a [vitest.dev/api/expect-typeof.html](https://vitest.dev/api/expect-typeof.html) for this specific case?

I haven't used that feature yet, and I do not know whether/how to proceed regarding testing of this type only feature.

**Comment by ST-DDT** [comment]
Would you consider this `!` breaking or not in the light of #1953?

- #1953

This is a user facing change, but it technically isn't anymore breaking than using the new node features after updating to them.

File: docs/guide/upgrading_v9/2685.md [hunk 1]
```diff
@@ -4,4 +4,13 @@ _This upgrade is an extension to_ [#1953](./1953.md)
 
 The helpers module now uses TS5 features, thus Faker requires TS5 for it to be used.
```
Comment (matthewmayer) [inline docs/guide/upgrading_v9/2685.md]: This seems slightly misleading, it makes it sound like you have to use TS to use Faker. But obviously you can use it with Javascript.

Perhaps 
> "The helpers module now uses TS5 features, so if you are using Faker with Typescript, you must use TS5."

**Review by matthewmayer — CHANGES_REQUESTED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Review by matthewmayer — APPROVED** [review]

**Review by ejcheng — APPROVED** [review]

**Pull request merged**